File: enclave-jvm-tester/djvm/djvm-tests-auxiliary/src/main/java/com/r3/conclave/jvmtester/djvm/testsauxiliary/tests/IterableTest.java
package com.r3.conclave.jvmtester.djvm.testsauxiliary.tests;

import java.util.ArrayList;
import java.util.function.Function;

public class IterableTest {
    public static class Create implements Function<Object, Iterable<String>> {

        @Override
        public Iterable<String> apply(Object o) {
            return new ArrayList<>();
        }
    }
}

File: enclave-jvm-tester/djvm/djvm-asserters/src/main/kotlin/com/r3/conclave/jvmtester/djvm/asserters/IterableTest.kt
package com.r3.conclave.jvmtester.djvm.asserters

import com.r3.conclave.jvmtester.api.TestAsserter
import org.assertj.core.api.Assertions.assertThat

class IterableTest {
    class Create : TestAsserter {
        override fun assertResult(testResult: ByteArray) {
            val result = String(testResult)
            assertThat(result).isEqualTo("sandbox.java.util.ArrayList.Itr")
        }
    }
}